sopel/coretasks.py

```diff
@@ -1127,12 +1150,19 @@ def sasl_mechs(bot, trigger):
 def _get_sasl_pass_and_mech(bot):
     password = None
     mech = None
+
     if bot.config.core.auth_method == 'sasl':
         password = bot.config.core.auth_password
         mech = bot.config.core.auth_target
     elif bot.config.core.server_auth_method == 'sasl':
         password = bot.config.core.server_auth_password
         mech = bot.config.core.server_auth_sasl_mech
+
+    if mech is None:
+        mech = 'PLAIN'
+    else:
+        mech = mech.upper()
```

Comment:
Sometimes I prefer one-liner for if/else:

```suggestion
    mech = 'PLAIN' if mech is None else mech.upper()
```

Or:

```suggestion
    mech = mech is None and 'PLAIN' or mech.upper()
```